packages/paste-core/layout/media-object/src/index.tsx

```diff
@@ -27,65 +35,68 @@ const MediaObject = React.forwardRef<HTMLElement, MediaObjectProps>(
-    children: PropTypes.node.isRequired,
-    verticalAlign: PropTypes.oneOf(['center', 'top']),
-  };
-}
 
-export interface MediaFigureProps extends Pick<BoxProps, 'as'> {
+MediaObject.propTypes = {
+  children: PropTypes.node.isRequired,
+  verticalAlign: PropTypes.oneOf(['center', 'top']),
+};
+
+export interface MediaFigureProps extends React.HTMLAttributes<any>, Pick<BoxProps, 'as' | 'element' | 'variant'> {
   align?: 'start' | 'end';
   children: NonNullable<React.ReactNode>;
   spacing?: Space;
 }
 
-const MediaFigure = React.forwardRef<HTMLElement, MediaFigureProps>(({as = 'span', children, align, spacing}, ref) => {
-  return (
-    <Box
-      as={as}
-      display="flex"
-      flexShrink={0}
-      marginLeft={align === 'end' ? spacing : undefined}
-      marginRight={align === 'start' ? spacing : undefined}
-      ref={ref}
-    >
-      {children}
-    </Box>
-  );
-});
+const MediaFigure = React.forwardRef<HTMLElement, MediaFigureProps>(
+  ({align, as = 'span', children, element = 'MEDIA_FIGURE', spacing, ...props}, ref) => {
+    return (
+      <Box
+        element={element}
+        {...safelySpreadBoxProps(props)}
+        as={as}
+        display="flex"
+        flexShrink={0}
+        marginLeft={align === 'end' ? spacing : undefined}
+        marginRight={align === 'start' ? spacing : undefined}
+        ref={ref}
+      >
+        {children}
+      </Box>
+    );
+  }
+);
 
 MediaFigure.displayName = 'MediaFigure';
+
 MediaFigure.defaultProps = {
   align: 'start',
   spacing: 'space0',
 };
-if (process.env.NODE_ENV === 'development') {
-  MediaFigure.propTypes = {
-    align: PropTypes.oneOf(['start', 'end']),
-    children: PropTypes.node.isRequired,
-    spacing: isMarginTokenProp,
-  };
-}
 
-// export type MediaBodyProps = Pick<BoxProps, 'as'>;
-export interface MediaBodyProps extends Pick<BoxProps, 'as'> {
+MediaFigure.propTypes = {
+  align: PropTypes.oneOf(['start', 'end']),
+  children: PropTypes.node.isRequired,
+  spacing: isMarginTokenProp,
+};
+
+export interface MediaBodyProps extends React.HTMLAttributes<any>, Pick<BoxProps, 'as' | 'element' | 'variant'> {
   children: NonNullable<React.ReactNode>;
 }
 
-const MediaBody = React.forwardRef<HTMLElement, MediaBodyProps>(({as = 'span', children}, ref) => {
-  return (
-    <Box as={as} flex={1} minWidth="size0" ref={ref}>
-      {children}
-    </Box>
-  );
-});
+const MediaBody = React.forwardRef<HTMLElement, MediaBodyProps>(
+  ({as = 'span', children, element = 'MEDIA_BODY', ...props}, ref) => {
+    return (
+      <Box element={element} {...safelySpreadBoxProps(props)} as={as} flex={1} minWidth="size0" ref={ref}>
+        {children}
+      </Box>
+    );
+  }
+);
 
 MediaBody.displayName = 'MediaBody';
-if (process.env.NODE_ENV === 'development') {
-  MediaBody.propTypes = {
-    children: PropTypes.node.isRequired,
-  };
-}
+
+MediaBody.propTypes = {
+  children: PropTypes.node.isRequired,
+};
 
 export {MediaObject, MediaFigure, MediaBody};
```

Comment:
Not that these MediaObject components have that much going on with them, but since they have 3 parts, would it be useful to split them up for clarity, the way we do for Modal, etc?